### PR TITLE
[opt](load) opt row_distribution performance in load (part II)

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

find_tablets

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

File: be/src/vec/sink/writer/vtablet_writer.cpp [hunk 1]
```diff
@@ -1786,22 +1792,27 @@ Status VTabletWriter::append_block(doris::vectorized::Block& input_block) {
                 return Status::NeedSendAgain("");
             }    // creating done
         } else { // not auto partition
-            for (int i = 0; i < num_rows; ++i) {
-                if (UNLIKELY(has_filtered_rows) && _block_convertor->filter_map()[i]) {
-                    continue;
+            std::vector<VOlapTablePartition*> partitions {num_rows, nullptr};
+            std::vector<bool> skip;
+            skip.resize(num_rows);
```
Comment (HappenLee) [inline be/src/vec/sink/writer/vtablet_writer.cpp:1797]: why not call call contruct like partitions，no need call `resize`

File: be/src/vec/sink/writer/vtablet_writer.cpp [hunk 2]
```diff
@@ -1737,34 +1740,37 @@ Status VTabletWriter::append_block(doris::vectorized::Block& input_block) {
             missing_map.reserve(partition_col.column->size());
 
             // try to find tablet and save missing value
-            for (int i = 0; i < num_rows; ++i) {
-                if (UNLIKELY(has_filtered_rows) && _block_convertor->filter_map()[i]) {
-                    continue;
+            std::vector<VOlapTablePartition*> partitions {num_rows, nullptr};
```
Comment (HappenLee) [inline be/src/vec/sink/writer/vtablet_writer.cpp:1743]: the code should be simple and same as not auto parttition

File: be/src/vec/sink/writer/vtablet_writer.cpp (hunk 1 above)
Comment (zclllyybb) [reply be/src/vec/sink/writer/vtablet_writer.cpp:1797]: `vector<bool>` is not same as other vector

**Pull request closed without merging**